### ISSUE TITLE
No mention of ServiceUI.EXE in PSADT manual

### ISSUE THREAD

**Issue by That-Annoying-Guy** [PSAppDeployToolkit/PSAppDeployToolkit#816]
ServiceUI.EXE is mentioned in the PSADT Discourse site but not PSADT manual.

**Describe the solution you'd like**
There is a post that describe how to use it:
https://discourse.psappdeploytoolkit.com/t/answer-on-running-silent-mode-with-serviceui/4654

Based on it, I propose to include something like the following in a future version of the PSADT manual:

**ServiceUI.exe** is a component from the Microsoft Deployment Toolkit. You can use ServiceUI.exe to Bring a SYSTEM Process to Interactive Mode. This is useful for creating interactive installations with Microsoft's Intune.

Here is an example for installation command:
`ServiceUI.exe -Process:explorer.exe Deploy-Application.exe -DeploymentType "Install"`

Here is an example for Uninstallation command:
`ServiceUI.exe -Process:explorer.exe Deploy-Application.exe -DeploymentType "Uninstall"`

Note: The `-Process:explorer` switch is used to determine where the install runs i.e. in the context of where explorer.exe is running (The user context), The `-DeploymentType` is optional, but will by default be the Install command if not specified.

**Comment by That-Annoying-Guy** [comment]
Then vote for an interactive install feature here: https://feedbackportal.microsoft.com/feedback/idea/bc1837a6-29c6-ed11-a81b-002248519701 

If you do, consider voting for users to be able to uninstall, too : https://feedbackportal.microsoft.com/feedback/idea/4b35db55-f5a7-ed11-a81b-000d3a0450e3

**Comment by fethiarras** [comment]
This tricks not works with Workspace One UEM MDM

**Comment by That-Annoying-Guy** [comment]
Therefore ServiceUI.EXE is specific to SCCM/Intune.
Maybe the makers of **Workspace One UEM MDM** have a similar utility.
If that is the case, please report back.

**Comment by fethiarras** [comment]
Yes with Workspace One UEM makers embedded a way to execute action in user context. But some actions must be deployed in system context and it's useful to make interactive actions in some cases

**Comment by That-Annoying-Guy** [comment]
So in Workspace One UEM MDM you can launch in the SYSTEM context or the User context.
But they don't have anything that will allow users to see GUI popups when running in the SYSTEM context.

You'd have to create something yourself like creating a scheduled task running in the user context or using [KelvinTegelaar](https://github.com/KelvinTegelaar)'s [RunAsUser](https://github.com/KelvinTegelaar/RunAsUser)

**Comment by fethiarras** [comment]
It's the same feature of this [https://psappdeploytoolkit.com/docs/reference/functions/Execute-ProcessAsUser](url) ?

**Comment by seanlillis** [comment]
If no user is logged on, we should add some logic to check if explorer.exe is running (user logged on) before invoking ServiceUI, otherwise launch Deploy-Application.exe without ServiceUI. Perhaps a ServiceUI launch script would be useful to include with the toolkit.

Ref: https://x.com/xenappblog/status/1761053837140341152?s=20

/cc @sintaxasn

**Comment by DanGough** [comment]
^ That’s the method I’ve used whenever I’ve needed to make an interactive install via Intune.

It’s also worth investigating this as an alternative to ServiceUI:

https://github.com/PhrozenIO/PowerRunAsSystem

**Comment by DanGough** [comment]
I've created a gist here with my proposal for a script to solve this issue - if anyone interested would like to give it a try and has any suggestions, please post them to gist page itself:

https://gist.github.com/DanGough/7ad3b9193ae66c0573fa5986c181c491
